src/cartesian/ReferenceLine.tsx

```diff
@@ -150,7 +138,19 @@ export const getEndPoints = (
 };
 
 export function ReferenceLine(props: Props) {
-  const { x: fixedX, y: fixedY, segment, xAxis, yAxis, shape, className, alwaysShow, clipPathId } = props;
+  const { x: fixedX, y: fixedY, segment, xAxisId, yAxisId, shape, className, alwaysShow } = props;
+
+  const [{ xAxisMap, yAxisMap, viewBox, clipPathId }] = useContext(ChartLayoutContext);
+  if (!clipPathId || !xAxisMap || !yAxisMap || !viewBox) {
+    return null;
+  }
+  const xAxis = xAxisMap[xAxisId];
+  const yAxis = yAxisMap[yAxisId];
+
+  warn(
+    xAxis != null,
+    `Could not find xAxis by id ${xAxisId} [${typeof xAxisId}]. Available xAxis are: ${Object.keys(xAxisMap)}`,
```

Comment:
is this an invariant or a warning? If we don't have an xAxis isnt that an error?

Reply:
The component will throw few lines below anyway, might just as well make it invariant.

Reply:
Done

Reply:
To confirm I wrote these tests on `master` branch:

```  test('throws an error when rendered without any axes', () => {
    render(
      <BarChart width={1100} height={250}>
        <ReferenceLine y={20} />
      </BarChart>,
    );
  });

  test('throws an error when rendered without YAxis', () => {
    render(
      <BarChart width={1100} height={250}>
        <XAxis />
        <ReferenceLine y={20} />
      </BarChart>,
    );
  });

  test('throws an error when rendered without XAxis', () => {
    render(
      <BarChart width={1100} height={250}>
        <YAxis />
        <ReferenceLine y={20} />
      </BarChart>,
    );
  });
```

And indeed they all do fail:

<img width="335" alt="image" src="https://github.com/recharts/recharts/assets/1100170/d4a3793c-d30b-4d5b-a308-321ac7ec6fd0">

<img width="795" alt="image" src="https://github.com/recharts/recharts/assets/1100170/3c4fe86b-11d6-4545-9d51-a1c98a62d7d2">

So a side effect of this PR is a better error message.

I wonder if there is ask to have the ReferenceLine work with one axis defined. I will browse through open issues later.